react/src/components/Chat/ChatCard.tsx

```diff
@@ -80,38 +83,61 @@ function useEndpoint(selectedEndpoint?: ChatCard_endpoint$key | null) {
   return { endpoint, setEndpoint } as const;
 }
 
+function createModelsURL(baseURL: string) {
+  const { origin, port, pathname: path } = new URL(baseURL.trim());
+  const host = port.length > 0 ? `${origin}:${port}` : origin;
+  const normalizedPath = path === '/' ? '/models' : `${path}/models`;
+
+  return new URL(normalizedPath, host).toString();
+}
+
 function useModels(
   provider: ChatProviderType,
   fetchKey: string,
-  endpoint?: ChatCard_endpoint$data | null,
+  baseURL?: string,
+  token?: string,
 ) {
+  const { message: appMessage } = App.useApp();
   const { data: modelsResult } = useSuspenseTanQuery<{
     data: Array<Model>;
   }>({
-    queryKey: ['models', fetchKey, endpoint?.endpoint_id],
-    queryFn: () => {
-      return endpoint?.url
-        ? fetch(
-            new URL(
-              provider.basePath + '/models',
-              endpoint?.url ?? undefined,
-            ).toString(),
-          )
-            .then((res) => res.json())
-            .catch((e) => ({ data: [] }))
-        : Promise.resolve({ data: [] });
+    queryKey: ['models', fetchKey, baseURL, token],
+    queryFn: async () => {
+      try {
+        if (baseURL) {
+          const url = createModelsURL(baseURL);
+          const authToken = token || provider.apiKey;
+          const res = await fetch(url, {
+            headers: {
+              Authorization: authToken ? `Bearer ${authToken}` : '',
+            },
+          });
+
+          if (!res.ok) {
+            throw new ChatModelError(res.status);
+          }
+
+          return await res.json();
+        }
+      } catch (error: any) {
+        // prevent to show the error message as failed fetching in the first time
+        if (fetchKey !== 'first') {
+          appMessage.error(`Error fetching models: ${error.message}`, 5);
+        }
+      }
+      return { data: [] };
     },
   });
 
-  const models = _.map(modelsResult?.data, (m) => ({
+  const models = map(modelsResult?.data, (m) => ({
     id: m.id,
     name: m.id,
   })) as BAIModel[];
 
   const selectedModelId = useMemo(
     () =>
       provider.modelId &&
-      _.includes(_.map(modelsResult?.data, 'id'), provider.modelId)
+      includes(map(modelsResult?.data, 'id'), provider.modelId)
```

Comment:
Is there any reason to use lodash functions without the `_.` prefix?

Reply:
It's my bad but using destructuring importing or loadsh-es are better practice (it's out of scope, we can discuss later)